violet-framework/src/main/java/com/horstmann/violet/workspace/sidebar/historypanel/HistoryPanel.java

```diff
@@ -0,0 +1,120 @@
+package com.horstmann.violet.workspace.sidebar.historypanel;
+
+import com.horstmann.violet.framework.injection.resources.ResourceBundleInjector;
+import com.horstmann.violet.workspace.sidebar.colortools.IColorChoiceBar;
+import com.horstmann.violet.workspace.sidebar.editortools.EditorToolsPanelUI;
+
+import java.awt.*;
+import java.awt.event.*;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+
+import com.horstmann.violet.framework.injection.resources.ResourceBundleInjector;
+import com.horstmann.violet.framework.injection.resources.annotation.ResourceBundleBean;
+import com.horstmann.violet.workspace.IWorkspace;
+import com.horstmann.violet.workspace.editorpart.IEditorPart;
+import com.horstmann.violet.workspace.editorpart.IEditorPartBehaviorManager;
+import com.horstmann.violet.workspace.editorpart.behavior.CutCopyPasteBehavior;
+import com.horstmann.violet.workspace.editorpart.behavior.UndoRedoCompoundBehavior;
+import com.horstmann.violet.workspace.sidebar.ISideBarElement;
+import com.horstmann.violet.workspace.sidebar.SideBar;
+
+/**
+ * Created by Marek on 06.12.2016.
+ */
+@ResourceBundleBean(resourceReference = SideBar.class)
+public class HistoryPanel extends JPanel implements ISideBarElement {
+
+    public HistoryPanel() {
+
+        ResourceBundleInjector.getInjector().inject(this);
+        this.ui= new HistoryPanelUI(this);
+        add(scrollPane);
+        list.setSize(100,50);
+        list.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        scrollPane.setSize(150,100);
+        add(scrollPane);
+        list.addMouseListener(new MouseAdapter() {
+            public void mousePressed(MouseEvent mouseEvent) {
+
+                if (mouseEvent.getClickCount() == 2) {
+                    int selectedIndex = list.getSelectedIndex();
+                    UndoRedoCompoundBehavior undoRedoBehavior = getUndoRedoBehavior();
+                    for (int i=0; i<list.getModel().getSize();i++){
+                        if (i >= selectedIndex){
```

Comment:
You should extract this "for (int i=0; i<list.getModel().getSize();i++){..}" to smaller private method to avoid nested code.
Make sure that you are using SonarLint, it allows only for triple nesting
But remember it is only my suggestion.